sectionsUtilityTool/src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionUtilityToolFilter.java

```diff
@@ -0,0 +1,171 @@
+package edu.umich.ctools.sectionsUtilityTool;
+
+import java.io.IOException;
+import java.util.Hashtable;
+import java.util.Properties;
+
+import javax.naming.Context;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import edu.umich.its.lti.utils.PropertiesUtilities;
+
+
+
+public class SectionUtilityToolFilter implements Filter {
+	private static final String OU_GROUPS = "ou=Groups";
+	private static final String PROVIDER_URL = "ldap://ldap.itd.umich.edu:389/dc=umich,dc=edu";
+	private static final String LDAP_CTX_FACTORY = "com.sun.jndi.ldap.LdapCtxFactory";
+	private static final String ASSETS_PATH = "/assets";
+	private static Log M_log = LogFactory.getLog(SectionUtilityToolFilter.class);
+	protected static final String SYSTEM_PROPERTY_FILE_PATH = "sectionsToolPropsPath";
+	protected static final String PROPERTY_CANVAS_ADMIN = "canvas.admin.token";
+	protected static final String PROPERTY_CANVAS_URL = "canvas.url";
+	protected static final String PROPERTY_TEST_URL = "test.url";
+	private static final String AUTH_GROUP = "its-canvas-sections";
+	private static final String TEST_USER = "testUser";
+	protected static Properties canvasProperties = null;
+	private boolean isTestUrlEnabled=false;
+
+	@Override
+	public void init(FilterConfig filterConfig) throws ServletException {
+		M_log.debug("Filter Init(): Called");
+		getCanvasCredentials();
+		
+	}
+
+	@Override
+	public void doFilter(ServletRequest request, ServletResponse response,
+			FilterChain chain) throws IOException, ServletException {
+		M_log.debug("doFilter: Called");
+		HttpServletRequest useRequest = (HttpServletRequest) request;
+		HttpServletResponse useResponse=(HttpServletResponse)response;
+		if(!checkForAuthorization(useRequest)) {
+		      useResponse.sendError(403);
+			return;
+		}
+		chain.doFilter(useRequest, response);
+	}
+
+	@Override
+	public void destroy() {
+		M_log.debug("detroy: Called");
+	}
+	protected void getCanvasCredentials() {
+		M_log.debug("getCanvasCredentials(): called");
+		String propertiesFilePath = System.getProperty(SYSTEM_PROPERTY_FILE_PATH);
+		if (!isEmpty(propertiesFilePath)) {
+		canvasProperties=PropertiesUtilities.getPropertiesObjectFromURL(propertiesFilePath);
+		}else {
+			M_log.error("File path for (sectionsToolProps.properties) is not provided");
```

Comment:
Process won't terminate. This method is invoked as part of init() so it will say canvasProperties=null